fixbackend/auth/permission_checker.py

```diff
@@ -0,0 +1,43 @@
+#  Copyright (c) 2024. Some Engineering
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from fastapi import HTTPException
+from fixbackend.auth.depedencies import AuthenticatedUser
+from fixbackend.auth.models import WorkspacePermission
+
+from logging import getLogger
+
+from fixbackend.workspaces.dependencies import UserWorkspaceDependency
+
+log = getLogger(__name__)
+
+
+class WorkspacePermissionChecker:
+    def __init__(self, required_permissions: WorkspacePermission):
+        self.required_permissions = required_permissions
+
+    async def __call__(self, user: AuthenticatedUser, workspace: UserWorkspaceDependency) -> bool:
```

Comment:
Not happy with this code:
- why do we need an authenticated user? We have the permissions in the JWT -> no database lookup required.
- why do we need to load the workspace? it is a path parameter -> no database lookup required.